### PR TITLE
API client script to create project assignment portioning

### PR DESCRIPTION
**Pull request by abelrischa**
# How to test
Sample files to start:
[create_project_portioned_assignment samples.zip](https://github.com/datasaur-ai/api-client-example/files/13886233/create_project_portioned_assignment.samples.zip)
replace the contents of the `documents` directory with the contents of that zip file

1. In the Datasaur appcreate a team with ~5 members
2. create any kind of project labeling project
  a. better to match the project kind with the sample files you'll be using
  b. the sample files I linked above are txts, so you can create a span or span+doc labeling project 
3. configure step 2 and 3 however you like
4. in step 4, assign the users and their roles, ex: 3 labelers, 2 reviewers
5. in step 5 configure things how you want
6. click the `<> View Script` button on the top-right
7. copy the contents of `JSON script (createProject)`
8. paste the contents to `create_project.json` in your directory
9. run the api_client as instructed in the readme

### CODE REVIEW

**Comment by ivanm-gdp** [comment]
ah, ga bisa 👀

**Comment by abelrischa** [comment]
oke gw rebase manual deh ya

**Comment by ivanm-gdp** [comment]
udah bisa nih reopen
tapi perlu resolve conflict 👀

File: create-project-async/src/portioned_assignment.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import glob
+from itertools import combinations
+import json
+from src.exceptions.invalid_options import InvalidOptions
+
+class AssignmentRole:
+  LABELER = "LABELER"
+  REVIEWER = "REVIEWER"
+  HYBRID = "LABELER_AND_REVIEWER"
```
Comment (ivanm-gdp) [inline create-project-async/src/portioned_assignment.py:9]: aaa
kenapa class 🤔 

ini maunya enum kan ya

Comment (ivanm-gdp) [reply create-project-async/src/portioned_assignment.py:9]: ohhh ternyata memang pakai class ya
cman extend enum
https://docs.python.org/3.11/howto/enum.html

```py
from enum import Enum

class AssignmentRole(Enum):
  ...
```

Comment (abelrischa) [reply create-project-async/src/portioned_assignment.py:9]: iya maksudnya jadi enum
gw baru ketemu ini juga sih wkwkwk
https://docs.python.org/3/library/enum.html

otw update

File: create-project-async/src/portioned_assignment.py [hunk 2]
```diff
@@ -18,16 +18,16 @@ def __init__(self, original_assignments: list[dict], multi_pass_prefix: str, sin
...
         self.reviewer_team_member_ids = []
         for assignment in self.original_assignments:
             team_member_id = assignment["teamMemberId"]
-            role: AssignmentRole = assignment["role"]
+            role: str = assignment["role"]
 
             # split reviewer and labeler assignees
             # labeler assignees include `LABELER_AND_REVIEWER` role
-            if (role == AssignmentRole.REVIEWER):
+            if (role == AssignmentRole.REVIEWER.value):
```
Comment (ivanm-gdp) [inline create-project-async/src/portioned_assignment.py:30]: this is fine ya, ini FYI aja (karena aku baru coba-coba jg

bisa compare begini jg sepeertinya
```py
if (AssignmentRole[role] == AssignmentRole.REVIEWER):
  # should also return True if role="REVIEWER"
  pass
```

image

syntax preference aja sih, not sure mana yang lebih python-ic

File: create-project-async/src/portioned_assignment.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    def create_new_assignments_by_documents(self, documents_path):
+        if (len(self.original_assignments) == 0):
+            return self.original_assignments
+
+        # splits single-pass and multi-pass files
+        multi_pass_file_names = []
+        single_pass_file_names = []
+        for filepath in glob.iglob(f"{documents_path}/*"):
+            file_name = filepath.split('/')[-1]
```
Comment (ivanm-gdp) [inline create-project-async/src/portioned_assignment.py:47]: `'/'`

kayanya better akses os.path.sep / os.sep

better yet, pakai os.path.split()
https://docs.python.org/3/library/os.path.html#os.path.split

Comment (ivanm-gdp) [reply create-project-async/src/portioned_assignment.py:47]: aku baru notice jg existing pakai hardcoded separator jg 🤔 
tidak platform independent

Comment (abelrischa) [reply create-project-async/src/portioned_assignment.py]: updated ya @ivanm-gdp 
https://github.com/datasaur-ai/api-client-example/pull/25/commits/922466924ecd8b2363ed7038150e84792c135c19

script yang existing di project.py juga gw ganti jadi pake `os.path.split`

Comment (ivanm-gdp) [reply create-project-async/src/portioned_assignment.py]: wah thankyou bel 🙏🏼

**Review by ivanm-gdp — APPROVED** [review]

**Comment by abelrischa** [comment]
rebased ke new branch separate from master
biar kalo mau push fixes kecil2 bisa langsung 👌

**Comment by abelrischa** [comment]
dari kak @rikoap apakah ada feedback kak?

**Comment by rikoap** [comment]
wait.. bentar lg yah...

**Review by rikoap — APPROVED** [review]

**Comment by rikoap** [comment]
Bisa dilanjut ya Bel @abelrischa

**Pull request merged**